DatadogLogs/Sources/Log/LogEventEncoder.swift

```diff
@@ -75,6 +75,8 @@ public struct LogEvent: Encodable {
         public var sourceType: String = "ios"
         /// The custom fingerprint supplied for this error, if any
         public var fingerprint: String?
+        /// Binary images needed to decode the provided stack (if any)
+        public var binaryImages: [BinaryImage]?
```

Comment:
change-request/ There are two problems with this code:
- Separation of concerns: We encode `DatadogInternal.BinaryImage` directly into log event sent to EvP. Because `BinaryImage` is considered an internal code, it can be changed at any time. An unaware change contributor may not even notice that altering it will break the contract with logs intake.
- The property is marked mutable, hence user may want to interact with and redact the `BinaryImage` through log event mapper. However, without [re-exporting the `DatadogInternal.BinaryImage` from `DatadogLogs`](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogRUM/Sources/RUMConfiguration.swift#L12) the compiler and / or autocompletion may be lost with dynamic lookup.

To address both, we should rather introduce a new type (`LogEvent.BinaryImage`?) that will mirror the `DatadogInternal.BinaryImage`. This way we achieve both: this model will prevent from accidentally breaking the contract due to the internal change and it will serve as first-class-citizen in `DatadogLogs` public interface.

Note, that this is exactly what we do with `BinaryImages` in RUM. They're [mapped here](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogRUM/Sources/DataModels/RUMDataModelsMapping.swift#L98-L113).